### ISSUE TITLE
SecConn[Read/Write]StateLimit Instantly hits limit - ap_get_scoreboard_worker method failing

### ISSUE THREAD

**Issue by afletch** [SpiderLabs/ModSecurity#843]
Apache 2.4.12
modsecurity 2.8.0

Setting either SecConnReadStateLimit or SecConnWriteStateLimit causes near immediate hit of the limit value (in line with every connection being counted), this is because every iteration started at apache2/mod_security2.c:1446/1447 is causing the count to be increased at 1466 (READ) and 1471 (WRITE). 

This appears to be due to line 1455 and the condition for Apache 2.4+ API differences:
`ws_record = ap_get_scoreboard_worker(sbh);`

Replacing this with:
`ws_record = ap_get_scoreboard_worker_from_indexes(i, j);`

Fixes the problem.
I would imagine the surrounding code also needs a tidy up.

**Comment by artefactio** [comment]
I believe I have the same problem with Apache 2.4.16 and ModSecurity 2.9.0. I'm using cPanel.

How to reproduce:
```
SecConnEngine "On"
SecConnReadStateLimit 50
SecConnWriteStateLimit 50
```

After Apache restart, every connection to server is blocked and thread count in apache error_log always show maximum number of threads - 256, for this is my Max Request Workers (MaxClients) setting.

Log example:

> [Wed Jan 20 07:55:38.439996 2016] [:warn] [pid 64146] ModSecurity: Access denied with code 400. Too many threads [256] of 50 allowed in READ state from x.x.x.x - Possible DoS Consumption Attack [Rejected]
> [Wed Jan 20 07:55:38.457756 2016] [:warn] [pid 64147] ModSecurity: Access denied with code 400. Too many threads [256] of 50 allowed in READ state from x.x.x.x - Possible DoS Consumption Attack [Rejected]
> [Wed Jan 20 07:55:38.461716 2016] [:warn] [pid 64148] ModSecurity: Access denied with code 400. Too many threads [256] of 50 allowed in READ state from x.x.x.x - Possible DoS Consumption Attack [Rejected]
> [Wed Jan 20 07:55:47.133584 2016] [:warn] [pid 64149] ModSecurity: Access denied with code 400. Too many threads [256] of 50 allowed in READ state from y.y.y.y - Possible DoS Consumption Attack [Rejected]
> [Wed Jan 20 07:55:47.869582 2016] [:warn] [pid 64150] ModSecurity: Access denied with code 400. Too many threads [256] of 50 allowed in READ state from z.z.z.z - Possible DoS Consumption Attack [Rejected]

These are legitimate requests; it blocked my IP also after performing only one web page refresh.

ModSec obviously gets number of threads wrong in my case.

Line 1455 of mod_security2.c also has ap_get_scoreboard_worker(sbh). I'm unwilling to modify and recompile, since this is a production server.

**Comment by zimmerle** [comment]
Fixed as part of #1340.
